datastructures/pieces/Bishop.cpp

```diff
@@ -1,39 +1,75 @@
 #include "Bishop.h"
+#include "../../detectors/CheckDetector.h"
 
 Bishop::Bishop(const Position &initialPosition, const Color &initialPlayer) : Piece(initialPosition, initialPlayer) {}
 
-std::vector<Position> Bishop::possibleMoves() {
+std::vector<Position> Bishop::possibleDirectMoves(Board& board) {
 
     std::vector<Position> positions;
 
     // top left to bottom right
     Position comparison = currentPosition;
+    Piece *potentialPiece;
     while(comparison.row < 8 && comparison.column > 'A'){
         comparison.row ++;
         comparison.column --;
+        potentialPiece = board.getPiece(comparison);
+        if(potentialPiece != nullptr && potentialPiece != this){
```

Comment:
Is it necessary to add the condition potentialPiece != this?

Also, you forgot to add the position to the positions vector if the if statement is not satisfied.

Reply:
And another thing, if the piece we see is the opponents, it should be added to the vector and then break out of the while.

Reply:
yeah, pretty sure that the `potentialPiece != this is redundant`, Ben had also pointed this out. However I handle the opponent piece logic in the next while. This first block sets the cursor to the top left and the second block moves it to the bottom right and handles pushing the positions.

Reply:
I might rework the diagonal logic a bit.